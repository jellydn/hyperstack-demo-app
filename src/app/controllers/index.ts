// naming and imports: name and import as you see fit
import Auth from './auth'
import { NotesController } from './notes'

const controllers = []
controllers.push(Auth)
controllers.push(NotesController)

export const AppController = {
  controllers,
}
