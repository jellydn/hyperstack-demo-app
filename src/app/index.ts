import type { Context } from 'hyperstack'
import { context } from 'hyperstack'
import type * as Models from './models'
import type * as Mailers from './mailers'

export const appContext = context as Context<typeof Models, typeof Mailers>
