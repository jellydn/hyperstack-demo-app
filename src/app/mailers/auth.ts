import { Mailer } from 'hyperstack'
import type { User } from '../models/user'

class AuthMailer extends Mailer {
  static defaults = {
    from: `Elle Postage <no-reply@elle-postage.example.com>`,
  }

  static forgotPassword(user: Partial<User>) {
    return this.mail({
      template: 'auth/forgot',
      message: {
        to: user.username,
      },
      locals: {
        resetToken: user.resetToken,
        name: user.name,
      },
    })
  }

  static sendWelcome(user: Partial<User>) {
    return this.mail({
      template: 'auth/welcome',
      message: {
        to: user.username,
      },
      locals: {
        verifyToken: user.emailVerificationToken,
        name: user.name,
      },
    })
  }
}

export { AuthMailer }
