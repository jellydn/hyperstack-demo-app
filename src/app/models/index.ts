export { Note } from './note'
export { User } from './user'
