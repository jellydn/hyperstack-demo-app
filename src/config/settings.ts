// general purpose settings, with no infrastructure loading, no assumptions. keep it simple.
import path from 'path'
export const root = path.join(__dirname, '..')
