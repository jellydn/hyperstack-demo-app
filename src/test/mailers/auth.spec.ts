import { test } from '@hyperstackjs/testing'
import { AuthMailer } from '../../app/mailers'
import { root } from '../../config/settings'
import { appContext } from '../../app'

const {
  mailers,
  serializers: { baseMailerSerialize },
} = test(root)
describe('mailers', () => {
  describe('welcome', () => {
    // a 'returns delivery' mailer test
    mailers('should send welcome', async () =>
      AuthMailer.sendWelcome({
        username: 'joe@example.com',
        name: 'joe',
      })
    )
    // a manual mailer test
    mailers('should send welcome: manual', async () => {
      // can grab from app context
      const { AuthMailer } = appContext.mailers()
      AuthMailer.clearDeliveries()
      await AuthMailer.sendWelcome({
        username: 'joe@example.com',
        name: 'joe',
      }).deliver()
      expect(AuthMailer.deliveries().map(baseMailerSerialize)).toMatchSnapshot()
    })
  })
})
