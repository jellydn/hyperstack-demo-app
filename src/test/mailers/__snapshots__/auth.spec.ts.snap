// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mailers welcome should send welcome 1`] = `
{
  "content": undefined,
  "contentType": "multipart/alternative",
  "contents": [
    undefined,
    "Welcome joe, you can now log in.
  Verify your account with the link below:

  http://localhost/verify?verifyToken=
",
    "<html>

<body>
  You can <a href="http://localhost:/verify?verifyToken=">verify your account</a>
</body>

</html>
",
  ],
  "envelope": {
    "from": "no-reply@elle-postage.example.com",
    "to": [
      "joe@example.com",
    ],
  },
  "from": "no-reply@elle-postage.example.com",
  "messageId": "test-redacted",
  "originalMessage": {
    "attachments": [],
    "from": "Elle Postage <no-reply@elle-postage.example.com>",
    "html": "<html>

<body>
  You can <a href="http://localhost:/verify?verifyToken=">verify your account</a>
</body>

</html>
",
    "subject": "Welcome joe",
    "text": "Welcome joe, you can now log in.
  Verify your account with the link below:

  http://localhost/verify?verifyToken=
",
    "to": "joe@example.com",
  },
  "response": "test-redacted",
  "subject": "Welcome joe",
  "to": [
    "joe@example.com",
  ],
}
`;

exports[`mailers welcome should send welcome: manual 1`] = `
[
  {
    "content": undefined,
    "contentType": "multipart/alternative",
    "contents": [
      undefined,
      "Welcome joe, you can now log in.
  Verify your account with the link below:

  http://localhost/verify?verifyToken=
",
      "<html>

<body>
  You can <a href="http://localhost:/verify?verifyToken=">verify your account</a>
</body>

</html>
",
    ],
    "envelope": {
      "from": "no-reply@elle-postage.example.com",
      "to": [
        "joe@example.com",
      ],
    },
    "from": "no-reply@elle-postage.example.com",
    "messageId": "test-redacted",
    "originalMessage": {
      "attachments": [],
      "from": "Elle Postage <no-reply@elle-postage.example.com>",
      "html": "<html>

<body>
  You can <a href="http://localhost:/verify?verifyToken=">verify your account</a>
</body>

</html>
",
      "subject": "Welcome joe",
      "text": "Welcome joe, you can now log in.
  Verify your account with the link below:

  http://localhost/verify?verifyToken=
",
      "to": "joe@example.com",
    },
    "response": "test-redacted",
    "subject": "Welcome joe",
    "to": [
      "joe@example.com",
    ],
  },
]
`;
